src/client/components/Divider.tsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import { css, SerializedStyles } from '@emotion/react';
+
+import { border, text } from '@guardian/src-foundations/palette';
+import { space } from '@guardian/src-foundations';
+import { textSans } from '@guardian/src-foundations/typography';
+import { from } from '@guardian/src-foundations/mq';
+
+type Sizes = 'full' | 'partial' | 'fit';
+type Spaces = 'tight' | 'loose';
+type Props = {
+  size?: Sizes;
+  spaceAbove?: Spaces;
+  displayText?: string;
+};
+
+const fullStypes = css`
+  margin-left: -10px;
+  width: calc(100% + 20px);
+`;
+const partialStyles = css`
+  margin-left: -10px;
+  width: 160px;
+`;
+
+const decideSpace = (
+  spaceAbove: Spaces,
+  displayText?: string,
+): SerializedStyles => {
+  switch (spaceAbove) {
+    case 'tight':
+      return displayText
+        ? css`
+            margin-top: -3px;
+          `
+        : css`
+            margin-top: ${space[6]}px;
+          `;
+    case 'loose':
+      return displayText
+        ? css`
+            margin-top: ${space[4]}px;
+            ${from.mobileMedium} {
+              margin-top: ${space[6]}px;
+            }
+          `
+        : css`
+            margin-top: ${space[12]}px;
+          `;
+  }
+};
+
+const decideSize = (size: Sizes) => {
+  switch (size) {
+    case 'fit':
+      return css`
+        :before {
+          margin-left: 0;
```

Comment:
What do you think about this being a copy of the version that we have ready to go into source (https://github.com/guardian/source/pull/1012) instead so we just have to drop in the css overrides for the behaviour and remove this once it's been added to the package